src/ethereum_clis/clis/geth.py

```diff
@@ -290,26 +368,71 @@ def consume_blockchain_test(
                 f"Unexpected exit code:\n{' '.join(command)}\n\n Error:\n{result.stderr}"
             )
 
-    @cache  # noqa
-    def consume_state_test_file(
+    # ------------------------------------ eof test -----------------------------------------------
+    def extract_int_from_subprocess_string(
+        self, s: str, substring_of_interest_prefix: str
+    ) -> int | None:
+        """Hacky method to extract relevant substring from string returned by subprocess execution."""  # noqa: E501
+        # get everything after this substring occurrence, then remove everything not in [0,9]
+        relevant_substring = re.sub(r"\D", "", s.split('total executed"=', 1)[1])
```

Comment:
I think it should be:
```suggestion
        relevant_substring = re.sub(r"\D", "", s.split(substring_of_interest_prefix, 1)[1])
```

Reply:
This one is failing after the change for me:
For `s` equal to `'INFO [03-17|17:33:22.741] Executed tests passed=4 "total executed"=4'`, `relevant_substring` becomes `44` when `substring_of_interest_prefix=='tests passed='`.

I think this function can be replaced with:
```python
m = re.search(r'tests passed=(\d+) "total executed"=(\d+)', s)
```
and then tests passed is `int(m.group(1))` and tests executed is `int(m.group(2))`.